File: packages/shared/types/competitor.ts
export interface OfferAnalysisRequest {
  offer: string
  niche: string
  city?: string
  goal?: string
}

export interface OfferAnalysisResult {
  strength_score: number
  whats_working: string[]
  whats_missing: string[]
  improved_offers: string[]
  recommended_hook_types: string[]
}

export interface CompetitorRequest {
  instagram_handle: string
  niche?: string
}

export interface PostSummary {
  caption_preview: string
  likes: number
  estimated_views: number
}

export interface CompetitorResult {
  handle: string
  posts_analyzed: number
  avg_likes: number
  top_content_types: string[]
  best_posts: PostSummary[]
  recommended_strategy: string
  strengths: string[]
  gaps: string[]
}

File: apps/api/app/models/competitor.py
from pydantic import BaseModel
from typing import Optional


class CompetitorRequest(BaseModel):
    instagram_handle: str
    niche: Optional[str] = None


class PostSummary(BaseModel):
    caption_preview: str
    likes: int
    estimated_views: int


class CompetitorResult(BaseModel):
    handle: str
    posts_analyzed: int
    avg_likes: float
    top_content_types: list[str]
    best_posts: list[PostSummary]
    recommended_strategy: str
    strengths: list[str]
    gaps: list[str]
